android/anc/src/main/java/org/smartregister/fhircore/anc/data/model/PatientModels.kt

```diff
@@ -106,5 +106,37 @@ data class PatientVitalItem(
   var bpds: String = "",
   var bpdsUnit: String = "",
   var pulse: String = "",
-  var pulseUnit: String = ""
-)
+  var pulseUnit: String = "",
+  var bmi: String = "",
+  var bmiUnit: String = ""
+) {
+  fun isWeightAndHeightAreValid(): Boolean {
+    return weight.isNotEmpty() &&
+      height.isNotEmpty() &&
+      weight.toDouble() > 0 &&
+      height.toDouble() > 0
+  }
+
+  fun isWeightAndHeightAreInMetricUnit(): Boolean {
+    return weightUnit.equals(UnitConstants.UNIT_WEIGHT_METRIC, true) &&
+      heightUnit.equals(UnitConstants.UNIT_HEIGHT_METRIC, true)
+  }
+
+  fun isWeightAndHeightAreInUSCUnit(): Boolean {
+    return weightUnit.equals(UnitConstants.UNIT_WEIGHT_USC, true) &&
+      heightUnit.equals(UnitConstants.UNIT_HEIGHT_USC, true)
+  }
+}
+
+object UnitConstants {
+  const val UNIT_WEIGHT_METRIC = "kg"
+  const val UNIT_HEIGHT_METRIC = "cm"
+  const val UNIT_CODE_WEIGHT_METRIC = "kg"
+  const val UNIT_CODE_HEIGHT_METRIC = "cm"
+  const val UNIT_BMI_METRIC = "kg/m2"
+  const val UNIT_WEIGHT_USC = "lb"
+  const val UNIT_HEIGHT_USC = "in"
+  const val UNIT_CODE_WEIGHT_USC = "[lb_av]"
+  const val UNIT_CODE_HEIGHT_USC = "[in_i]"
+  const val UNIT_BMI_USC = "kg/m2" // can be set as lb/in2
```

Comment:
confused by this, isn't this lb/in2 as in the comment?

Reply:
I was also confused, but as per original ticket
https://github.com/opensrp/fhircore/issues/521
unit for bmi found kg/m2, over googling too, but keeping comment if in future still thats need it can be change from here

Reply:
ah I gotcha, very interesting, that's a standard BMI unit regardless of the input units, thank you!